File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/data/database/mapper/RealmMediaMapper.java
package com.lattis.ellipse.data.database.mapper;

import androidx.annotation.NonNull;

import com.lattis.ellipse.data.database.base.AbstractRealmDataMapper;
import com.lattis.ellipse.data.database.model.RealmMedia;
import com.lattis.ellipse.domain.model.Media;

import javax.inject.Inject;

/**
 * Created by raverat on 4/6/17.
 */

public class RealmMediaMapper extends AbstractRealmDataMapper<Media, RealmMedia> {

    @Inject
    public RealmMediaMapper() {}

    @NonNull
    @Override
    public RealmMedia mapIn(@NonNull Media media) {
        RealmMedia realmMedia = new RealmMedia();
        realmMedia.setId(media.getId());
        realmMedia.setUrl(media.getUrl());
        return realmMedia;
    }

    @NonNull
    @Override
    public Media mapOut(@NonNull RealmMedia realmMedia) {
        Media media = new Media();
        media.setId(realmMedia.getId());
        media.setUrl(realmMedia.getUrl());
        return media;
    }

}

File: velolabs_repos/lattis_android_v2/data/src/main/java/com/lattis/data/database/mapper/RealmMediaMapper.kt
package com.lattis.data.database.mapper

import com.lattis.data.database.base.AbstractRealmDataMapper
import com.lattis.data.database.model.RealmMedia
import com.lattis.domain.models.Media
import javax.inject.Inject

/**
 * Created by raverat on 4/6/17.
 */

class RealmMediaMapper @Inject
constructor() : AbstractRealmDataMapper<Media, RealmMedia>() {

    override fun mapIn(media: Media): RealmMedia {
        val realmMedia = RealmMedia()
        realmMedia.id = media.id
        realmMedia.url = media.url
        return realmMedia
    }

    override fun mapOut(realmMedia: RealmMedia): Media {
        val media = Media()
        media.id = realmMedia.id
        media.url = realmMedia.url
        return media
    }

}
